src/lib.rs

```diff
@@ -1180,6 +1180,34 @@ impl Bump {
         }
     }
 
+    /// Like `alloc_slice_copy`, but does not panic in case of allocation failure.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// let bump = bumpalo::Bump::new();
+    /// let x = bump.try_alloc_slice_copy(&[1, 2, 3]);
+    /// assert_eq!(x, Ok(&mut[1, 2, 3] as &mut [_]));
+    ///
+    ///
+    /// let bump = bumpalo::Bump::new();
+    /// bump.set_allocation_limit(Some(4));
+    /// let x = bump.try_alloc_slice_copy(&[1, 2, 3, 4, 5, 6]);
+    /// assert_eq!(x, Err(bumpalo::AllocErr)); // too big
+    /// ```
+    #[inline(always)]
+    pub fn try_alloc_slice_copy<T>(&self, src: &[T]) -> Result<&mut [T], AllocErr>
+    where
+        T: Copy,
+    {
+        let layout = Layout::for_value(src);
+        let dst = self.try_alloc_layout(layout)?.cast::<T>();
+        let result = unsafe { slice::from_raw_parts_mut(dst.as_ptr(), src.len()) };
```

Comment:
I don't think this is safe because it is creating a reference of potentially uninitialized data. I think this needs to use [`std::ptr::copy_nonoverlapping`](https://doc.rust-lang.org/nightly/std/ptr/fn.copy_nonoverlapping.html) instead.

Reply:
Thank you, good point. `core::ptr::copy_nonoverlapping` for the no_std case.